graphql_compiler/ast_manipulation.py

```diff
@@ -95,3 +95,15 @@ def get_only_selection_from_ast(ast, desired_error_type):
                                      .format(ast_name))
 
     return selections[0]
+
+
+def get_ast_with_non_null_stripped(ast):
+    """Strip a NonNullType layer around the ast if there is one, return the underlying ast."""
+    if isinstance(ast, NonNullType):
+        stripped_ast = ast.type
+        if isinstance(stripped_ast, NonNullType):
+            raise AssertionError(u'NonNullType may not be wrapped around another NonNullType, '
```

Comment:
nit: Hitting this error message would be somewhat confusing – the user's reaction would be "of course it can be, I just gave you an AST that does that". I would reword this to emphasize that for this code, this situation was _unexpected_, since that is obviously true.